packages/openchs-android/src/views/common/IndividualProfile.js

```diff
@@ -31,6 +33,40 @@ class IndividualProfile extends AbstractComponent {
         style: PropTypes.object
     };
 
+    getMobileNoFromObservation() {
+        var i;
+        for (i = 0; i < this.props.individual.observations.length; i++) {
+            const observation = this.props.individual.observations[i];
+            const concept = observation.concept;
+            const keyValue = concept.recordValueByKey('primary_contact') || concept.recordValueByKey('contact_number');
+            if (keyValue === 'yes') {
+                const value = observation.getReadableValue();
+                return value.toString();
+            }
```

Comment:
Move this logic to Observation model in avni-models repo so it's reusable and also so the implementation details are not exposed.